### DISCUSSION TITLE
Start different programs

### DISCUSSION THREAD

**Discussion by pagasam** [siku2/hass-vzug#48] [Device Support]
Hello everyone, Hello @siku2

Thank you for this HomeAssistant integration.
Will it be possible in the future to send commands to start the different programs via the integration?
This has been implemented in the V-ZUG app. -> Pictures
In future, I would like to start the fully loaded machine when there is sufficient photovoltaic surplus. 

![20240528_120357000_iOS](https://github.com/siku2/hass-vzug/assets/112263662/73473f15-7d5d-467e-9721-5ed4ba4ccc87)
![20240528_120407000_iOS](https://github.com/siku2/hass-vzug/assets/112263662/f2dcde05-3ce3-41ef-a2fd-b4071cfa1479)

**Comment by siku2** [comment]
@pagasam this is already partially implemented in the code. The reason I didn't continue with it is because my devices still require manual confirmation to start the program.

Can you actually start your dishwasher remotely using the app? In my case I always need to open the door, confirm the program, and then close it again.

**Comment by pagasam** [comment]
@siku2 
In my case, the AdoraDish V4000 door has to be open so that programs can be sent to the appliance via the V-ZUG app. As you said.
The other appliances (AdoraDry V6000 and AdoraWash V4000) do not even display the programs to be selected in the V-ZUG app.

Have you been able to test sending a program function from HomeAssistant (without the V-ZUG app) with a dryer, dishwasher or washing machine?

**Reply by siku2** [reply]
Yes, I tried this back when I reverse-engineered their app. I have a dishwasher as well as a washing machine and it worked with both (that's not surprising though because I ripped the implementation from the app).
But because there was no way to start the machines fully remotely I didn't bother adding something like a service to the integration.

Do you have a use-case where this semi-automatic integration is useful?

**Reply by pagasam** [reply]
@siku2 
I can imagine such a semi-automatic integration in connection with the solar production surplus.

For example, you could calculate the best start time of the machines for the following day depending on the solar surplus. 
From this, the start delay (or the exact start time) could be calculated and sent to the dryer, washing machine or tumble dryer together with the appropriate program. 
The machine door would then have to be closed manually the evening before.

What do you think of this approach? 

Have you had any contact with V-ZUG regarding the non-existent function to start the machines remotely?

**Reply by siku2** [reply]
I agree that this would be a good use case. Have you tested whether you can change the start delay after the door is closed?

> Have you had any contact with V-ZUG regarding the non-existent function to start the machines remotely?

No, I haven't tried to contact V-ZUG at all.

**Reply by danijelst** [reply]
Just in case you need additional info about the REST API for this, I found this link https://hack.opendata.ch/project/225

E.g.: https://192.168.1.100/hh?command=getProgram

Returns: `[{"id":54,"name":"Intensiv","status":"active","duration":{"act":4905},"energySaving":{"set":false},"partialload":{"set":false},"rinsePlus":{"set":false},"dryPlus":{"set":false}}]`

There are also set methods but I didn't try yet.

**Reply by siku2** [reply]
Thanks for sharing this link, I'll take a closer look later today.

FYI: Both `getProgram` and `setProgram` are already in my implementation:
https://github.com/siku2/hass-vzug/blob/5a67d177f3f04688c135598fc89554cb90aca2fd/custom_components/vzug/api/__init__.py#L588-L613
